### ISSUE TITLE
test: comprehensive test suite — 132 tests across all modules

### ISSUE THREAD

**Issue by sanskarpan** [sanskarpan/Optimization#14]
## Overview
Complete test coverage for all 6 optimization modules, growing from 80 original tests to **132 passing tests**.

### TestOptimizers (25 tests)
Core tests for all first-order and Adam-family optimizers: numeric updates, reset behavior, direction checks, gradient clipping.

### TestOptimizersExtended (7 new tests)
- `test_rmsprop_reset` — verifies accumulator `v` cleared to zeros
- `test_rmsprop_numeric` — step-by-step manual calculation
- `test_adam_weight_decay` — weight decay reduces parameter magnitude
- `test_nadam_numeric` — NAdam direction differs from Adam
- `test_amsgrad_convergence` — converges on simple 1-D problem
- `test_clip_gradients_empty` — empty gradient list returns `[]`
- `test_clip_gradients_zero_max_norm` — zero max_norm clips to all-zeros

### TestLearningRateSchedules (15 tests) + TestLearningRateSchedulesExtended (14 new tests)
- WarmRestartLR validation guards, period doubling across two cycles, reset
- Exponential decay both modes, step decay, cosine annealing reset
- OneCycleLR: end value, anneal phase, reset
- ReduceLROnPlateau: absolute threshold, min_lr floor, reset

### TestLineSearch (11 tests) + TestLineSearchExtended (6 new tests)
- Reversed/equal endpoint validation for golden section
- Non-square A and wrong-length b for exact quadratic
- Backtracking max_iter exhaustion behavior

### TestSecondOrder (10 tests) + TestSecondOrderExtended (11 new tests)
- `test_newton_step_singular_middle_pivot` — BUG-C2 regression test
- BFGS H_inv symmetry after update, curvature guard rejects negative s·y
- L-BFGS deque eviction at capacity m
- CG 1-D steepest descent, beta=0 for near-zero old gradient
- All four second-order classes have `reset()`

### TestConstrained (10 tests) + TestConstrainedExtended (9 new tests)
- Lagrange λ sign, convergence at returned point
- KKT via two-inequality formulation
- Barrier method with two constraints, infeasible start terminates
- Simplex projection: z=2.0, empty input, nonpositive z raises

### TestGlobalOptimization (8 tests) + TestGlobalOptimizationExtended (5 new tests)
- SA 1-D convergence
- GA pop_size < 3 raises
- PSO large bounds stable, v_max clamping
- DE 1-D convergence

### TestEdgeCases (1 test)
- All optimizers produce zero step for zero gradient

## Result
```
132 passed in 0.14s
```
